### PR TITLE
[25408] - Retry Implementation for Get Queries.

### PR DESCRIPTION
**Pull request by AlejandroOrozco**
## Ticket
- [25408]

## Description
- Add retry for get queries  at SecureStorage.

## Commit message
-  fix: Add retry for keychain operations, trying to prevent 25300 keychain error code [25408]

## Testing
- [x] Added unit tests
- [x] Ran unit tests successfully
- [ ] Added documentation for public APIs and/or Wiki

### CODE REVIEW

File: TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/SecureStorageTests.swift [hunk 1]
```diff
@@ -163,6 +163,53 @@ class SecureStorageTests: XCTestCase {
...
+    let expectation = expectation(description: "Test multiple tries of storage.Get")
+    guard let expectedData = "data".data(using: .utf8) else {
+      XCTFail("failed to encrypt data")
+      return
+    }
+
+    let key = "key"
+    var data: Data?
+    keychain.error = TestError.operationFailed
+    keychain.keys = [expectedData as AnyObject, expectedData as AnyObject , expectedData as AnyObject]
```
Comment (AuthyApps) [inline TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/SecureStorageTests.swift:178]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/b8f1e4c4694ac4732d318a7f2b21e7a8ef80870b/TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/SecureStorageTests.swift#L178"></span>There should be no space before and one after any comma.<br />
<code>comma</code> <code>SecureStorageTests.swift:178</code></td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`93.83%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`87.23%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by mejiagarcia — APPROVED** [review]

File: TwilioVerifySDK/TwilioSecurity/Sources/Storage/SecureStorage.swift [hunk 2]
```diff
@@ -125,4 +129,27 @@ extension SecureStorage: SecureStorageProvider {
       }
     }
   }
+
+  private func retry<T>(
+    tries: Int = 3,
+    block: () throws -> T
```
Comment (yafuquen) [inline TwilioVerifySDK/TwilioSecurity/Sources/Storage/SecureStorage.swift:135]: Should we include a [delay](https://github.com/twilio/twilio-verify-android/blob/main/security/src/main/java/com/twilio/security/crypto/AndroidKeyManager.kt#L143) to give some time?

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`93.43%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/Keychain.swift | **`87.29%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifySDK/TwilioSecurity/Sources/Storage/SecureStorage.swift [hunk 3]
```diff
@@ -69,7 +73,7 @@ extension SecureStorage: SecureStorageProvider {
     Logger.shared.log(withLevel: .info, message: "Getting \(key)")
     let query = keychainQuery.getData(withKey: key)
     do {
-      let result = try keychain.copyItemMatching(query: query)
+      let result: AnyObject = try retry { try keychain.copyItemMatching(query: query) }
```
[the remark this answers is not in the document]
Comment (AlejandroOrozco) [reply TwilioVerifySDK/TwilioSecurity/Sources/Storage/SecureStorage.swift]: Agree, I moved the retry implementation to the Keychain.swift. Only for Sign & CopyItemMatching for now. Thank you.

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`93.40%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/Keychain.swift | **`87.36%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by yafuquen — APPROVED** [review]

**Review by mejiagarcia — APPROVED** [review]

**Pull request merged**